assets/js/react-shim.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * React shim for ensuring all modules share the same instance of React.
```

Comment:
Since this is from such a long time ago, I'd be interested in revisiting whether or not this is really necessary. WP core and Gutenberg don't do this, but I'm wondering if that's because WP core doesn't use any dependencies that import react?